specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json

```diff
@@ -3446,52 +3992,55 @@
           {
             "name": "name",
             "in": "path",
-            "description": "Name of the app.",
-            "required": true,
-            "type": "string"
-          },
-          {
-            "name": "id",
-            "in": "path",
-            "description": "Deployment ID.",
+            "description": "Name of web app",
             "required": true,
             "type": "string"
           },
           {
             "name": "instanceId",
             "in": "path",
-            "description": "ID of a specific scaled-out instance. This is the value of the name property in the JSON response from \"GET api/sites/{siteName}/instances\"",
+            "description": "Id of web app instance",
             "required": true,
             "type": "string"
           },
           {
-            "$ref": "#/parameters/subscriptionIdParameter"
-          },
-          {
+            "name": "MSDeploy",
+            "in": "body",
+            "description": "Details of MSDeploy operation",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/MSDeploy"
```

Comment:
Here's the source that's used to generate that Swagger:
`        // PUT api/sites/{name}/extensions/MSDeploy
        // PUT api/sites/{name}/slots/{slotname}/extensions/MSDeploy
        // PUT api/sites/{name}/instances/{instanceId}/extensions/MSDeploy
        // PUT api/sites/{name}/slots/{slotname}/instances/{instanceId}/extensions/MSDeploy
        /// <summary>
        /// Invoke the MSDeploy web app extension as pass-through API
        /// </summary>
        /// <param name="subscriptionId">Your Azure subscription ID. This is a GUID-formatted string (e.g. 00000000-0000-0000-0000-000000000000).</param>
        /// <param name="resourceGroupName">Name of resource group</param>
        /// <param name="name">Name of web app</param>
        /// <param name="slot">Name of web app slot. If not specified then will default to production slot.</param>
        /// <param name="instanceId">Id of web app instance</param>
        /// <returns>MSDeploy status</returns>
        /// <response code ="201">Deployment is scheduled</response>
        /// <response code ="409">Another deployment is in progress</response>
        [Route(UriElements.Csm.SitesResource + UriElements.Csm.NameTemplateParamWithConstraint +
            UriElements.Csm.MSDeployExtensionResource)]
        [Route(UriElements.Csm.SitesResource + UriElements.Csm.NameTemplateParamWithConstraint +
            UriElements.Csm.SlotsResource + UriElements.Csm.SlotTemplateParamWithConstraint +
            UriElements.Csm.MSDeployExtensionResource)]
        [Route(UriElements.Csm.SitesResource + UriElements.Csm.NameTemplateParamWithConstraint +
            UriElements.Csm.SiteInstancesResource + UriElements.Csm.SiteInstanceIdResource +
            UriElements.Csm.MSDeployExtensionResource)]
        [Route(UriElements.Csm.SitesResource + UriElements.Csm.NameTemplateParamWithConstraint +
            UriElements.Csm.SlotsResource + UriElements.Csm.SlotTemplateParamWithConstraint + UriElements.Csm.SiteInstancesResource + UriElements.Csm.SiteInstanceIdResource +
            UriElements.Csm.MSDeployExtensionResource)]
        [OperationIdResolver("CreateInstanceMSDeployOperation", path: UriElements.Csm.SitesResource + UriElements.Csm.NameTemplateParamWithConstraint +
            UriElements.Csm.SiteInstancesResource + UriElements.Csm.SiteInstanceIdResource +
            UriElements.Csm.MSDeployExtensionResource)]
        [OperationIdResolver("CreateInstanceMSDeployOperation", path: UriElements.Csm.SitesResource + UriElements.Csm.NameTemplateParamWithConstraint +
            UriElements.Csm.SlotsResource + UriElements.Csm.SlotTemplateParamWithConstraint +
            UriElements.Csm.SiteInstancesResource + UriElements.Csm.SiteInstanceIdResource + UriElements.Csm.MSDeployExtensionResource)]
        [RequestType(typeof(MSDeploy), Name = "MSDeploy", Description = "Details of MSDeploy operation", Required = true)]
        [ResponseType(typeof(MSDeployStatus))]
        [AzureOperationSettings(longRunning: true)]
        [HttpPut]
        public HttpResponseMessage CreateMSDeployOperation(string subscriptionId, string resourceGroupName, string name, string slot = null, string instanceId = null)
        {
            var pathAndQuery = UriElements.Csm.MSDeployExtensionResourceType + Request.RequestUri.Query;
            var response = this.CurrentController.InvokeSiteExtension(subscriptionId, resourceGroupName, name, slot, pathAndQuery, Request, instanceId: instanceId, viaExtensionsRoute: true);
            SiteExtensionPostResponseOperation(subscriptionId, resourceGroupName, name, slot, response);
            return response;
        }
`
Please tell me the magic annotation I need to put on this to make this problem go away.  ;)